ml/src/main/scala/org/apache/spark/ml/feature/RapidsPCA.scala

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.feature
+
+import org.apache.hadoop.fs.Path
+
+import org.apache.spark.ml._
+import org.apache.spark.ml.linalg._
+import org.apache.spark.ml.linalg.distributed.RapidsRowMatrix
+import org.apache.spark.ml.param._
+import org.apache.spark.ml.util._
+import org.apache.spark.sql._
+import org.apache.spark.sql.functions.{col, udf}
+import org.apache.spark.sql.types.StructType
+
+trait RapidsPCAParams extends PCAParams {
+  /**
+   * Whether to apply mean centering to the input data.
+   *
+   * @group param
+   */
+  final val meanCentering: BooleanParam = new BooleanParam(this, "meanCentering", "whether to apply mean centering")
+  setDefault(meanCentering, true)
+
+  /** @group getParam */
+  def getMeanCentering: Boolean = $(meanCentering)
+
+  /**
+   * Whether to use GEMM to compute the covariance matrix.
+   *
+   * @group param
+   */
+  final val useGemm: BooleanParam =
+    new BooleanParam(this, "useGemm", "whether to use GEMM to compute the covariance matrix")
+  setDefault(useGemm, true)
+
+  /** @group getParam */
+  def getUseGemm: Boolean = $(useGemm)
+
+  /**
+   * The GPU id to use.
+   *
+   * @group param
+   */
+  private[ml] final val gpuId: IntParam = new IntParam(this, "gpuId", "the GPU id to use")
```

Comment:
Curious, how is this set in a distributed fashion?  For example, I'm running on a node with 4 GPUs and 4 executors.  How does each instance get the appropriately assigned ID?
```suggestion
  private[ml] final val gpuId: IntParam = new IntParam(this, "gpuId", "the GPU ID to use")
```